api/src/processing/services/soundcloud.js

```
@@ -113,9 +113,17 @@ export default async function(obj) {
 
     if (!file) return { error: "fetch.empty" };
 
+    const title = json.title.trim();
+    const artist = json.user.username.trim();
     const fileMetadata = {
-        title: json.title.trim(),
-        artist: json.user.username.trim(),
+        title,
+        album: json.publisher_metadata.album_title?.trim() ?? title,
```

i don't think the fall back to song title makes sense here, wouldn't it be misleading most of the time?